dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/maintenance/jdbc/JdbcMaintenanceStore.java

```diff
@@ -304,6 +304,30 @@ public int deleteSoftDeletedTrackedEntityInstances()
         return result;
     }
 
+    @Override
+    public void prunePeriods()
+    {
+        String sql = "with period_ids(periodid) as (\n"
+            + "  select distinct periodid from completedatasetregistration\n"
```

Comment:
Looks good. Just curious, should we remove the line break / `\n`? We don't typically use this is SQL statements, there _might_ some issues with different platforms, and even ORMs don't typically format the SQL they execute.